eng/scripts/Language-Settings.ps1

```diff
@@ -0,0 +1,93 @@
+$Language = "rust"
+$LanguageDisplayName = "Rust"
+$PackageRepository = "crates.io"
+$packagePattern = "Cargo.toml"
+$MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/main/_data/releases/latest/rust-packages.csv"
+$GithubUri = "https://github.com/Azure/azure-sdk-for-rust"
+$PackageRepositoryUri = "https://crates.io/crates"
+
+function Get-AllPackageInfoFromRepo ([string] $ServiceDirectory) {
+  $allPackageProps = @()
+  Push-Location $RepoRoot
+  try {
+    $searchPath = Join-Path $RepoRoot 'sdk' -Resolve
+
+    if ($ServiceDirectory -and $ServiceDirectory -ne 'auto') {
+      $searchPath = Join-Path 'sdk' $ServiceDirectory
+    }
+
+    $packages = cargo metadata --format-version 1
+    | ConvertFrom-Json -AsHashtable
+    | Select-Object -ExpandProperty packages
+    | Where-Object { $_.manifest_path.StartsWith($searchPath) }
+
+    $packageManifests = @{}
+    foreach ($package in $packages) {
+      if ($package.manifest_path -replace '\\', '/' -match '/sdk/([^/]+)/') {
+        $package.ServiceDirectoryName = $Matches[1]
+      }
+      else {
+        # ignore manifests that are not in a service directory
+        continue
+      }
+
+      $package.RelativePath = (Split-Path $package.manifest_path -Parent).Replace($RepoRoot, "").SubString(1)
+      $package.DependentPackages = @()
```

Comment:
Just keep in mind you will need to update this once Scott's eng/common changes goes through.  https://github.com/Azure/azure-sdk-tools/pull/8878

Reply:
I'll have to manually pull in eng/common.   It'll give me an opportunity to uptake the new property name and callback then